src/utils.jl

```diff
@@ -11,3 +11,26 @@ maybe_real(x::Complex) = iszero(imag(x)) ? real(x) : x
 
 maybe_only(x) = x
 maybe_only(x::Tuple{T}) where {T} = only(x)
+
+
+#Find the edge in the list of edges which goes from source to dest (use the negative index if the edge is reversed)
+function find_edge(edges::Vector{NamedDimEdge{Tuple}}, source::Tuple, dest::Tuple)
+  for i = 1:length(edges)
+      if((src(edges[i]) == source && dst(edges[i]) == dest) || (dst(edges[i]) == source && src(edges[i]) == dest))
+          return i
+      end
+  end
+  return 0
+end
+
+#Get subset of edges involving a specific vertex
+function find_edges_involving_vertex(edges::Vector{NamedDimEdge{Tuple}}, vertex::Tuple)
```

Comment:
If the input was a graph instead of an edge list, you could just use: `Ref(vertex) .=> neighbors(g, vertex)` for this. Is there a reason you would have an edge list but not the actual graph?

Reply:
Ah I see yes this is much neater way to do that. The latest commit has this changed.